server/app/services/rpc/node_service_pod_handler.rb

```diff
@@ -1,21 +1,39 @@
 module Rpc
   class NodeServicePodHandler
+    include Logging
+
+    @@lru_cache ||= LruRedux::ThreadSafeCache.new(1000)
```

Comment:
Usually class variables are a bit code smell. Just look how you need to access this in specs. If we really need to have this at class level maybe then just class instance variable with proper accessor method?

But I think cache would be more predictable if we just create cache object per grid instance.